components/popover/index.js

```diff
@@ -100,27 +96,22 @@ class Popover extends Component {
 			return;
 		}
 
-		const { content } = this.nodes;
-		if ( ! content ) {
+		if ( ! this.contentNode.current ) {
 			return;
 		}
 
+		// Without the setTimeout, the dom node is not being focused
+		// Related https://stackoverflow.com/questions/35522220/react-ref-with-focus-doesnt-work-without-settimeout-my-example
+		const focusNode = ( domNode ) => setTimeout( () => domNode.focus() );
```

Comment:
Where are we canceling this timeout if the component is unmounted? Why aren't we using [`withSafeTimeout`](https://github.com/WordPress/gutenberg/tree/master/components/higher-order/with-safe-timeout) ?

Reply:
I did use `withSafeTimeout` originally but It breaks the design in a super weird way. (see 66b34440947efedcedff6eb38775c82e1f01ac81) The usage of the higher order component even if it's not doing anything to the DOM was creating some weird z-index issues, especially visible on mobile.

Reply:
What about: 

>Where are we canceling this timeout if the component is unmounted?

Reply:
This is already done in master